smart-contract-verifier/smart-contract-verifier-http/src/routers/solidity.rs

```diff
@@ -83,7 +83,12 @@ impl SolidityRouter {
             compilers_threads_semaphore,
         );
         compilers.load_from_dir(&dir).await;
-        let client = SolidityClient::new(compilers);
+        let mut client = SolidityClient::new(compilers);
+        if let Some(extensions) = settings.extensions {
+            client = client.with_middleware(
+                sig_provider_extension::SigProvider::new(extensions.sig_provider).await?,
```

Comment:
there is a problem, that if we enable this for all `solidity, sourcify, vyper`, then we'll have 3 instances of the middleware == 3 connections. not sure right now what's the best way to fix that

Reply:
I personally cannot see ways to fix that easily for now. If you also don't I suggest to merge the PR as it is, and think about that later. Maybe we may create a separate issue for that